### PR TITLE
[CELEBORN-874] Enrich Fetch log

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

As title

### Why are the changes needed?

As title

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/network/client/TransportClient.java [hunk 1]
```diff
@@ -120,16 +120,27 @@ public void fetchChunk(
...
-
-    StreamChunkSlice streamChunkSlice = new StreamChunkSlice(streamId, chunkIndex, offset, len);
     StdChannelListener listener =
         new StdChannelListener(streamChunkSlice) {
+          @Override
+          public void operationComplete(Future<? super Void> future) throws Exception {
+            if (logger.isDebugEnabled()) {
+              logger.debug(
+                  "Sending fetch chunk {} to {} operation completed",
+                  ((StreamChunkSlice) this.requestId).toString(),
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/client/TransportClient.java:138]: unnecessary toString()

File: common/src/main/java/org/apache/celeborn/common/network/client/TransportClientFactory.java [hunk 2]
```diff
@@ -176,6 +176,9 @@ public TransportClient createClient(
         }
       }
       clientPool.clients[clientIndex] = internalCreateClient(resolvedAddress, decoder);
+      logger.debug(
+          "local client channel address {}",
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/client/TransportClientFactory.java:180]: Please enrich the message, it's confusing what happened here on reading message.

File: common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java [hunk 3]
```diff
@@ -233,6 +233,14 @@ public void channelActive() {}
   @Override
   public void channelInactive() {
     if (numOutstandingRequests() > 0) {
+      if (logger.isDebugEnabled() && outstandingFetches.size() > 0) {
+        for (Map.Entry<StreamChunkSlice, FetchRequestInfo> entry : outstandingFetches.entrySet()) {
+          StreamChunkSlice key = entry.getKey();
+          logger.debug("outstandingFetches: {}", key);
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java:239]: ditto, supply the context in message

File: common/src/main/java/org/apache/celeborn/common/network/client/TransportClientFactory.java (hunk 2 above)
Comment (cfmcgrady) [reply common/src/main/java/org/apache/celeborn/common/network/client/TransportClientFactory.java]: removed, and added details local channel address when the client return.

File: common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java (hunk 3 above)
Comment (cfmcgrady) [reply common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java]: updated.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala [hunk 4]
```diff
@@ -243,6 +246,16 @@ class FetchHandler(val conf: CelebornConf, val transportConf: TransportConf)
       client.getChannel.writeAndFlush(new ChunkFetchSuccess(req.streamChunkSlice, buf))
         .addListener(new GenericFutureListener[Future[_ >: Void]] {
           override def operationComplete(future: Future[_ >: Void]): Unit = {
+            if (log.isDebugEnabled) {
+              if (future.isSuccess()) {
+                logInfo(
+                  s"Sending ChunkFetchSuccess operation succeeded, chunk ${req.streamChunkSlice}")
+              } else {
+                logError(
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala:254]: I think we should always print the error log, even it debug is disabled.

Comment (cfmcgrady) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala]: updated.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main/0.3

**Pull request closed without merging**